python/pyspark/mlv2/summarizer.py

```diff
@@ -0,0 +1,112 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+from pyspark.mlv2.util import aggregate_dataframe
+
+
+class SummarizerAggState:
+    def __init__(self, input_array):
+        self.min_values = input_array.copy()
+        self.max_values = input_array.copy()
+        self.count = 1
+        self.sum_values = np.array(input_array.copy())
+        self.square_sum_values = np.square(input_array.copy())
+
+    def update(self, input_array):
+        self.count += 1
+        self.sum_values += input_array
+        self.square_sum_values += np.square(input_array)
+        self.min_values = np.minimum(self.min_values, input_array)
+        self.max_values = np.maximum(self.max_values, input_array)
+
+    def merge(self, state):
+        self.count += state.count
+        self.sum_values += state.sum_values
+        self.square_sum_values += state.square_sum_values
+        self.min_values = np.minimum(self.min_values, state.min_values)
+        self.max_values = np.maximum(self.max_values, state.max_values)
+        return self
+
+    def to_result(self, metrics):
+        result = {}
+
+        for metric in metrics:
+            if metric == "min":
+                result["min"] = self.min_values.copy()
+            if metric == "max":
+                result["max"] = self.max_values.copy()
+            if metric == "sum":
+                result["sum"] = self.sum_values.copy()
+            if metric == "mean":
+                result["mean"] = self.sum_values / self.count
+            if metric == "std":
+                if self.count <= 1:
+                    raise ValueError(
+                        "Standard deviation evaluation requires more than one row data."
+                    )
+                result["std"] = np.sqrt(
```

Comment:
nit: why not reusing `torchmetrics` here?

Reply:
We need to evaluate std values over feature arrays, I think using summarizer will be more efficient.